api/types.go

```diff
@@ -823,6 +872,11 @@ func (pin *Pin) IsRemotePin(pid peer.ID) bool {
 	return true
 }
 
+// ExpiredAt returns whether the pin has expired at the given time.
+func (pin *Pin) ExpiredAt(t time.Time) bool {
+	return !(pin.ExpireAt.IsZero() || pin.ExpireAt.After(t))
```

Comment:
add an unexported constant `unixZero` for quicker comparison and check that t is not `unixZero` in which case it will not be expired. This is an extra safeguard in case someday someone ends up setting ExpireAt to Unix(0) when parsing a protobuf.